### DISCUSSION TITLE
Finder vs working printers?

### DISCUSSION THREAD

**Discussion by Pecillo** [moonglow/FlashForge_Marlin#80] [Q&A]
Hi

Is there a difference in the control boards that prevent this from working on a Finder?

/Adam

**Comment by moonglow** [comment]
Hi @Pecillo !

FF Finder 1 in theory will works too ( because they use STM32F4xx MCU ), but i do not know that pinout that board has. Dreamer/Inverter and etc that supported by my project has two different motherboard type and Finder has third type of MB ( and it looks much more different ). So i need to grab full pinout from that board and test my firmware before publish it. For other printers users donate me their motherboards so i test it all locally. IF someone will donate me Finder 1 motherboard i will try to support it too.

**Reply by Pecillo** [reply]
Hi @moonglow 

Well, I don't have a spare board. But I do have another (non FF) printer that should arrive today.
So I could open the Finder and assist in mapping the pins and testing a firmware once I have printer that I can use while the Finder acts as an test subject.

But if I do stumble over an extra Finder I'll gladly donate a card for this purpose. But for now, what can I do to help?

/Adam

**Reply by moonglow** [reply]
It be not so easy to do it remotely, but we can try. In few words we need to recreate board schematics, maps all pins and all mandatory peripheral IC ( as example for Dreamer: external eeprom, led rgb controller, thermocouple controller, digital potentiometers, sd cards and etc ).  First good step is to made hi resolution photo of board top and bottom side with good light and sharpness need to understand what a difference between other boards.

**Comment by Pecillo** [comment]
Hi

Here is a batch of images... Good enough to start with?

/Adam

![IMG_6249](https://user-images.githubusercontent.com/17779661/142511179-1d5752c0-9635-441a-bb5e-b0a4fb598fdd.JPG)
![IMG_6250](https://user-images.githubusercontent.com/17779661/142511181-25cd04a1-7dd8-4760-bdfb-10f2d1308fb8.JPG)
![IMG_6251](https://user-images.githubusercontent.com/17779661/142511182-20d95731-6844-482d-a149-e63bc64997fa.JPG)
![IMG_6252](https://user-images.githubusercontent.com/17779661/142511183-9a6a0753-6830-4f35-8cb9-cd0d3f918e30.JPG)
![IMG_6253](https://user-images.githubusercontent.com/17779661/142511184-c3d4f22d-7d8e-4a9b-a543-be90e1c2bcf1.JPG)
![IMG_6254](https://user-images.githubusercontent.com/17779661/142511185-aeef5244-9fba-416d-a2fd-341f816783b4.JPG)
![IMG_6255](https://user-images.githubusercontent.com/17779661/142511187-f3c5834c-5258-49ef-a665-9270dc63ff58.JPG)

**Reply by moonglow** [reply]
Now need to map all needed pins from pin config to that board, i think TFT interface will be same, but need to check all other pins from `pins_FF_MOTHERBOARD.h`.

**Reply by moonglow** [reply]
Hi @Pecillo ! Any news with board pin map ?

**Reply by Pecillo** [reply]
Sorry... Life got in the way of this... And printer has been in use.

I'll get the card out again and see if I can do the mapping.

**Comment by NoSadBeHappy** [comment]
I know this is an old post, but I would be willing to contribute to this.

I have one finder that has problems with the motor controllers, and another one that is functional. How do you know what the pins are called as referred in `pins_FF_MOTHERBOARD.h`? (ex: where does `PE2` come from for the `X_MIN_PIN`) 

Thanks!

**Comment by Mesteery** [comment]
Hello I have this board too (Finder 12v) and I did all the mapping (except wifi and touchscreen) and get it working with Klipper:
- PF15 EN WiFi power
- Extruder EN PE4 STEP PE6 DIR PC13 
- digipot E MCP4018 SDA PE5 SCL PF12
- digipot Z         SDA PF7 SCL PF12
- digipot Y         SDA PB9 SCL PF12
- digipot X         SDA PG14 SCL PF12
- Led pca9632 SDA PB1 SCL PF12
- Temp Ads1118 SPI3 mosi PB5 miso PB4 sclk PB3 cs PB6
- Servo CTRL PA2
- Probe PG8
- Switch Runout PG7
- Buzzer PB0
- On off button : PB12
- Fan PWM PC0
- Ps_on PA1
- X max PE2
- Y max PE3
- Heater PA3
- Y stepper EN PE0 STEP PB8 DIR PB7
- Z max PF10
- X stepper EN PG15 STEP PG13 DIR PG12
- Z stepper EN: PF6 STEP PF8 DIR PF9

<details>
  <summary> and here is my Klipper config if anyone is interested</summary>

I used Klipper from here https://github.com/mesteery/klipper thanks to work of dockterj https://github.com/dockterj/klipper

```cfg

[include fluidd.cfg]
[include macros.cfg]

[board_pins]
aliases: B_PB7=PB7

[exclude_object]

[virtual_sdcard]
path: ~/FlashforgeFinder_data/gcodes

[output_pin psu]
pin: PA1
value: 0
shutdown_value: 0

[pwm_cycle_time buzzer]
pin: PB0
value: 0
shutdown_value: 0
cycle_time: 0.001

[pca9632 led_strip]
color_order: BGRW
scl_pin: PF12
sda_pin: PB1

[filament_switch_sensor runout]
switch_pin: !PG7

[servo probe_arm]
pin: PA2
initial_angle: 26
maximum_servo_angle: 90

[probe]
x_offset: -7.6
y_offset: -25.8
#z_offset: 3
pin: PG8
speed: 2.0
lift_speed: 4.0
deactivate_on_each_sample: False
activate_gcode:
    SET_SERVO SERVO=probe_arm ANGLE=61
    G4 P300
deactivate_gcode:
    SET_SERVO SERVO=probe_arm ANGLE=26       
    #SET_SERVO SERVO=probe_arm ANGLE=26 ENABLE=0
    G4 P300

[safe_z_home]
home_xy_position: 85, 98.3
z_hop: 5

[bed_mesh]
horizontal_move_z: 5
mesh_min: 7, 9
mesh_max: 148, 121
probe_count: 5, 5
algorithm: bicubic
fade_start: 1
fade_end: 10
fade_target: 0

[bed_screws]
screw1: 28, 11
screw2: 128, 11
screw3: 78.5, 146

[screws_tilt_adjust]
screw1: 86.1, 147
screw1_name: center rear screw
screw2: 35.8, 36.8
screw2_name: left front screw
screw3: 135.8, 36.8
screw3_name: right front screw
screw_thread: CW-M3


[gcode_macro POWER_OFF_PRINTER]
gcode:
  {action_call_remote_method(
    "set_device_power", device="printer", state="off"
  )}

[delayed_gcode delayed_printer_off]
initial_duration: 0.
gcode:
  {% if printer.idle_timeout.state == "Idle" %}
    POWER_OFF_PRINTER
  {% endif %}

[idle_timeout]
gcode:
  M84
  TURN_OFF_HEATERS
  M106 S0
  UPDATE_DELAYED_GCODE ID=delayed_printer_off DURATION=60

[stepper_x]
step_pin: PG13
dir_pin: PG12
enable_pin: !PG15
microsteps: 16
rotation_distance: 34
endstop_pin: !PE2
position_endstop: 163
position_max: 163
homing_speed: 50

[stepper_y]
step_pin: PB8
dir_pin: !B_PB7
enable_pin: !PE0
microsteps: 16
rotation_distance: 34
endstop_pin: !PE3
position_endstop: 147
position_max: 147
homing_speed: 50

[stepper_z]
step_pin: PF8
dir_pin: PF9
enable_pin: !PF6
microsteps: 16
rotation_distance: 8
#endstop_pin: !PF10
#position_endstop: 140
endstop_pin: probe:z_virtual_endstop
position_max: 140
position_min: 0
homing_speed: 50

[ads1118]
sensor_pin: PB6
spi_software_sclk_pin: PB3
spi_software_miso_pin: PB4
spi_software_mosi_pin: PB5

[extruder]
max_extrude_only_distance: 101.0
min_extrude_temp: 0
step_pin: PE6
dir_pin: PC13
enable_pin: !PE4
microsteps: 16
rotation_distance: 32.77309124
nozzle_diameter: 0.400
filament_diameter: 1.750
heater_pin: PA3
sensor_type: ads1118_typek
sensor_pin: ads1118:pin_0
control: pid
pid_Kp: 20.720
pid_Ki: 0.785
pid_Kd: 136.753
min_temp: 0
max_temp: 260

[mcp4018 x_axis_pot]
scl_pin: PF12
sda_pin: PG14
wiper: 118
scale: 127

[mcp4018 y_axis_pot]
scl_pin: PF12
sda_pin: PB9
wiper: 118
scale: 127

[mcp4018 z_axis_pot]
scl_pin: PF12
sda_pin: PF7
wiper: 40
scale: 127

[mcp4018 e_axis_pot]
scl_pin: PF12
sda_pin: PE5
wiper: 118
scale: 127

[fan]
pin: PC0

[mcu]
serial: /dev/serial/by-id/usb-Klipper_stm32f407xx_…-if00
restart_method: command

[printer]
kinematics: cartesian
max_velocity: 300
max_accel: 3000
max_z_velocity: 5
max_z_accel: 100
```
</details>

**Reply by valik1017** [reply]
Hello!

But for Dreamer NX?

Thanks!
